sql/core/src/main/scala/org/apache/spark/sql/expressions/WindowSpec.scala

```diff
@@ -113,7 +132,25 @@ class WindowSpec private[sql](
     between(RangeFrame, start, end)
   }
 
-  private def between(typ: FrameType, start: Long, end: Long): WindowSpec = {
+  /**
+   * Defines the frame boundaries, from `start` (inclusive) to `end` (inclusive). And specifies
+   * potential exclude type
+   *
+   * Both `start` and `end` are relative from the current row. For example, "0" means "current row",
+   * while "-1" means one off before the current row, and "5" means the five off after the
+   * current row.
+   *
+   * @param start boundary start, inclusive.
+   *              The frame is unbounded if this is the minimum long value.
+   * @param end boundary end, inclusive.
+   *            The frame is unbounded if this is the maximum long value.
+   * @since 1.4.0
+   */
+  def rangeBetweenExclude(start: Long, end: Long, exclude: String = ""): WindowSpec = {
```

Comment:
We already have `rowsBetween` which doesn't specify exclude type. So I think we don't need to have default value for `exclude`.